src/methods/get.js

```diff
@@ -554,22 +554,21 @@ export function GetTemplateObject(realm: Realm, templateLiteral: BabelNodeTempla
 }
 
 export function GetFromArrayWithWidenedNumericProperty(realm: Realm, arr: ArrayValue, P: string | SymbolValue): Value {
-  let type = Value;
-
   if (typeof P === "string") {
-    // these are safe methods to allow, as they either return a new array or are a method
-    // that doesn't mutate global state or affect that of the values passed in as arguments
-    if (P === "map" || P === "slice" || P === "filter" || P === "concat" || P === "indexOf" || P === "reverse") {
-      let proto = arr.$GetPrototypeOf();
-      invariant(proto instanceof ObjectValue);
-      return OrdinaryGet(realm, proto, P, arr);
-    }
+    let proto = arr.$GetPrototypeOf();
```

Comment:
Since this function is only called if isIntrinsicAndHasWidenedNumericProperty is true for arr, can this ever be different from realm.intrinsics.ArrayPrototype? Should there be some sort of invariant.

Reply:
Good point.

Reply:
That's true, I've made those changes too :)